aws/resource_aws_s3_bucket_inventory.go

```diff
@@ -0,0 +1,445 @@
+package aws
+
+import (
+	"fmt"
+	"log"
+	"strings"
+	"time"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/s3"
+	"github.com/hashicorp/terraform/helper/resource"
+	"github.com/hashicorp/terraform/helper/schema"
+	"github.com/hashicorp/terraform/helper/validation"
+)
+
+func resourceAwsS3BucketInventory() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceAwsS3BucketInventoryPut,
+		Read:   resourceAwsS3BucketInventoryRead,
+		Update: resourceAwsS3BucketInventoryPut,
+		Delete: resourceAwsS3BucketInventoryDelete,
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"bucket": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+			"name": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ForceNew:     true,
+				ValidateFunc: validateMaxLength(64),
+			},
+			"enabled": {
+				Type:     schema.TypeBool,
+				Default:  true,
+				Optional: true,
+			},
+			"filter": {
+				Type:     schema.TypeList,
+				Optional: true,
+				MaxItems: 1,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"prefix": {
+							Type:     schema.TypeString,
+							Optional: true,
+						},
+					},
+				},
+			},
+			"destination": s3BucketInventoryDestinationSchema(),
```

Comment:
Minor nitpick: We are preferring to statically define the full schema inline in all its nested glory. This is for a few various reasons (future plans for automated code/documentation generation, linting, etc.), but not yet documented on the Terraform documentation website surrounding provider development. Hopefully it will be soon. 👍